### ISSUE TITLE
Soft deletes ignored in hasMany()->whereHas()

### ISSUE THREAD

**Issue by alistairreynolds** [ddzobov/laravel-pivot-softdeletes#2]
If I use a whereHas query like the one below, it returns soft deleted results

```
$this->hasMany(SiteApplication::class)
     ->whereHas('tags', function (Builder $query) {
         $query->whereIn('system_name', ['OUTLINE_APPLICATION', 'FULL_APPLICATION']);
     })
     ->get();```

**Comment by ddzobov** [comment]
Thank you!
Fixed in 1.1.1

**Comment by alistairreynolds** [comment]
Thanks! But now getting an error for ambigious deleted_at column:

```
SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'deleted_at' in where clause is ambiguous (SQL: select `id`, `site_id`, `decision`, `appeal_decision`, `appeal_submitted_date`, `appeal_decision_date`, `submitted_date`, `decision_date` from `site_applications` where `site_applications`.`site_id` = 73239 and `site_applications`.`site_id` is not null and exists (select * from `tags` inner join `site_application_tag` on `tags`.`id` = `site_application_tag`.`tag_id` where `deleted_at` is null and `site_applications`.`id` = `site_application_tag`.`site_application_id` and `system_name` in (OUTLINE_APPLICATION, FULL_APPLICATION) and `site_application_tag`.`deleted_at` is null and `tags`.`deleted_at` is null) and `site_applications`.`deleted_at` is null order by `is_primary_application` desc, GREATEST(coalesce(decision_date, 0), coalesce(appeal_decision_date, 0), coalesce(submitted_date, 0), coalesce(appeal_submitted_date, 0)) DESC)```

**Comment by ddzobov** [comment]
> Thanks! But now getting an error for ambigious deleted_at column:
> 
> ```
> SQLSTATE[23000]: Integrity constraint violation: 1052 Column 'deleted_at' in where clause is ambiguous (SQL: select `id`, `site_id`, `decision`, `appeal_decision`, `appeal_submitted_date`, `appeal_decision_date`, `submitted_date`, `decision_date` from `site_applications` where `site_applications`.`site_id` = 73239 and `site_applications`.`site_id` is not null and exists (select * from `tags` inner join `site_application_tag` on `tags`.`id` = `site_application_tag`.`tag_id` where `deleted_at` is null and `site_applications`.`id` = `site_application_tag`.`site_application_id` and `system_name` in (OUTLINE_APPLICATION, FULL_APPLICATION) and `site_application_tag`.`deleted_at` is null and `tags`.`deleted_at` is null) and `site_applications`.`deleted_at` is null order by `is_primary_application` desc, GREATEST(coalesce(decision_date, 0), coalesce(appeal_decision_date, 0), coalesce(submitted_date, 0), coalesce(appeal_submitted_date, 0)) DESC)```
> ```

Just published 1.1.2, check please

**Comment by alistairreynolds** [comment]
All good and working happily now 👍
